examples/multisig/methods/guest/src/bin/zk_multisig.rs

```diff
@@ -0,0 +1,345 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![no_main]
+
+use hex::FromHex;
+use k256::ecdsa::{RecoveryId, Signature, VerifyingKey};
+use risc0_zkvm::guest::env;
+use rs_merkle::{algorithms::Sha256, Hasher, MerkleProof, MerkleTree};
+use sha3::{Digest, Keccak256};
+
+risc0_zkvm::guest::entry!(main);
+
+const TRUSTED_MERKLE_ROOT: &str =
+    "1962fbc17a4440d383f544724e9a943773c0cc6eade89b66b724c31cf33d3f62";
+
+pub fn main() {
+    let signature_one_input: String = env::read(); // <ecdsa_sig_address_one + recovery_id>
+    let signature_two_input: String = env::read(); // <ecdsa_sig_address_two + recovery_id>
+    let merkle_proof_one_input: String = env::read(); // <proof_hashes + index_of_leaf>
+    let merkle_proof_two_input: String = env::read(); // <proof_hashes + index_of_leaf>
+    let signed_message_input: String = env::read(); // <signed_message>
+    let all_leaves_input: String = env::read(); // <merkle_leaves>
+
+    let all_leaves_bytes =
+        Vec::<u8>::from_hex(all_leaves_input.trim()).expect("Failed to decode hex string");
+
+    let leaves: Vec<[u8; 32]> = all_leaves_bytes
+        .chunks_exact(32)
+        .map(|chunk| {
+            let array: [u8; 32] = chunk.try_into().expect("Failed to convert chunk to array");
+            array
+        })
+        .collect();
+
+    let merkle_tree = MerkleTree::<Sha256>::from_leaves(&leaves);
+    let new_root_hex = merkle_tree
+        .root_hex()
+        .ok_or("couldn't get the merkle root")
+        .unwrap();
+    let new_root = merkle_tree.root().ok_or("couldn't get the merkle root");
+    let root_hash = match new_root {
+        Ok(hash) => hash,
+        Err(err) => {
+            println!("Error: {:?}", err);
+            return;
+        }
+    };
+    let merkle_root_log_message = "Merkle Root: ".to_owned() + &new_root_hex;
+    env::log(&merkle_root_log_message);
+
+    if !new_root_hex.eq(TRUSTED_MERKLE_ROOT) {
+        let error_message =
+            "Error, merkle root hash is incorrect. Expected".to_owned() + &TRUSTED_MERKLE_ROOT;
+        env::log(&error_message);
+        return;
+    }
+
+    if merkle_proof_one_input.eq(&merkle_proof_two_input) {
+        let error_message = "Error, merkle proofs cannot be the same!";
+        env::log(&error_message);
+        return;
+    }
+
+    /*
+     * Checking that the first merkle proof is valid
+     */
```

Comment:
Please avoid using C-style comments, use `//` instead

Reply:
Thank you for the feedback. It's my first time writing Rust so still learning the ropes.